files/etc/ublue-update.d/system/00-system-update.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+from subprocess import run
+from json import loads, load
+from json.decoder import JSONDecodeError
+import os
+
+
+def check_for_rebase():
+    # initialize variables
+    default_image_ref = []
+    current_image_ref = []
+    default_image_tag = ""
+    image_tag = ""
+
+    try:
+        with open("/usr/share/ublue-os/image-info.json") as f:
+            image_info = load(f)
+            default_image_ref = image_info["image-ref"].split(":")
+            default_image_tag = image_info["image-tag"]
+    except (FileNotFoundError, KeyError):
+        print("uBlue image info file does not exist")
+        return False, ""
+
+    status_cmd = [
+        "rpm-ostree",
+        "status",
+        "--pending-exit-77",
+        "-b",
+        "--json",
+    ]
+    status_out = run(status_cmd, capture_output=True)
+    if status_out.returncode != 0:
+        print(status_out.stdout.decode("utf-8"))
+        return False, ""
+
+    try:
+        current_image_ref = (
+            loads(status_out.stdout)["deployments"][0]["container-image-reference"]
+            .replace(
+                "ostree-unverified-registry:", "ostree-unverified-image:docker://"
+            )  # replace shorthand
+            .split(":")
+        )
+        if current_image_ref == default_image_ref:
+            return False, ""
```

Comment:
When on a signed image this should return False. But it doesn't happen because
current_image_ref contains `['ostree-image-signed', 'docker', '//ghcr.io/ublue-os/bluefin', '39']`
and
default_image_ref contains `['ostree-image-signed', 'docker', '//ghcr.io/ublue-os/bluefin']`
So it will never returns False

Reply:
resolved https://github.com/ublue-os/ublue-update/pull/81/commits/7953962aca971a0d54439c5240746e872b358efc